server/src/main/java/org/opensearch/plugins/PluginSubject.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.identity.AbstractSubject;
+import org.opensearch.identity.NamedPrincipal;
+import org.opensearch.identity.tokens.AuthToken;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.security.Principal;
+
+/**
+ * PluginSubject is passed to plugins within createComponents and does not require
+ * an auth token to authenticate
+ *
+ * @opensearch.api
```

Comment:
```suggestion
 * @opensearch.experimental
```

Reply:
Removed this class altogether in favor of only having a Noop in the core. The IdentityPlugin can define its own implementation of Subject meant for IdentityAwarePlugins.